### ISSUE TITLE
Proposal: TimePicker should be split into single digits again.

### ISSUE THREAD

**Issue by shaheedmalik** [microsoft/microsoft-ui-xaml#905]
# Proposal: [TimePicker should be split back into into single digits again.] 

## Rationale

* {It is time consuming to select the wanted minutes to times in apps the use the time control}
* {If I want to select 35 minutes for example, I will have to either stroll 35 times or 25 times to select that number, versus with separate controls, I will scroll only 7 times or 9 times}
* {etc}

## Scope

| Capability | Priority |
| :---------- | :------- |
| This proposal will allow developers to accomplish W | Must |
| This proposal will allow end users to accomplish X | Should |
| This proposal will allow developers to accomplish Y | Could |
| This proposal will allow end users to accomplish Z | Won't |

**Comment by mdtauk** [comment]
I have to disagree about splitting the controls into single digits.  But an editable Time Picker?

![image](https://user-images.githubusercontent.com/7389110/59816297-7b12c480-9313-11e9-838d-55bae9225a3e.png)

**Comment by shaheedmalik** [comment]
> I have to disagree about splitting the controls into single digits. But an editable Time Picker?
> 
> ![image](https://user-images.githubusercontent.com/7389110/59816297-7b12c480-9313-11e9-838d-55bae9225a3e.png)

It may look nice on paper but it isn't functional.

**Comment by mdtauk** [comment]
> > I have to disagree about splitting the controls into single digits. But an editable Time Picker?
> > ![image](https://user-images.githubusercontent.com/7389110/59816297-7b12c480-9313-11e9-838d-55bae9225a3e.png)
> 
> It may look nice on paper but it isn't functional.

![image](https://user-images.githubusercontent.com/7389110/59878098-582df200-939f-11e9-9fc4-595c5c80f412.png)

Making it editable means you can type in the time, which is the easiest way to do it with Keyboard, and still able to press to bring up the picker controls for touch.  Also there is a proposal for an Editable Date Picker too, so this would fit well with that.

The problem with breaking apart the two digit combos, comes with making the meaning clear for the reading of hours, minutes.
![image](https://user-images.githubusercontent.com/7389110/59877805-aabade80-939e-11e9-9487-05bc7b2768bb.png)

Adding seconds into the mix doesn't help matters.
![image](https://user-images.githubusercontent.com/7389110/59878503-60d2f800-93a0-11e9-831a-956ea05b3789.png)

**Comment by shaheedmalik** [comment]
Typing the time doesn't help someone using a tablet.

**Comment by mdtauk** [comment]
If the proposal is accepted, I would not want it to become the default.  I think it should be a property of the control. ``TimePicker.SplitDigits  = true;``

**Comment by dpierangeli** [comment]
I'm with @mdtauk . Yes, the split digits could help the tablet users (could because the only real tablet OS out there - iOS - uses the compact digit and no one has complained) but it can be a secondary and optional mode, to not trouble those apps that are geared to the desktop.

**Comment by kikisaints** [comment]
I had an internal proposal similar to this a while ago. It was basically a proposal to change TimePicker to three ComboBoxes "linked" together seamlessly. That meant that inherited properties like IsEditable would come along, but the base control would operate as you'd expect with keyboard, mouse and touch - because ComboBox works nicely with all three.

![combobox_timepicker_vid](https://user-images.githubusercontent.com/6964369/59938489-fa82bf80-9409-11e9-9df2-101d32d9189c.gif)

**Comment by mdtauk** [comment]
@kikisaints  that's is an interesting approach. The chevrons do add some bulk, which could be massaged with a design similar to the ones I explored.

It doesn't quite address the issue in the proposal, as they wanted to avoid long scrolls for the minutes, by having the two digits of the minutes separate.

But like the proposal for an Editable Date Picker, an Editable Time Picker, would fill the control suite out nicely.

Does **IsEditable** become a property on the existing controls, or should they be new controls?

----

What was the conclusion of that internal proposal?

**Comment by kikisaints** [comment]
Ah I misunderstood the original proposal a bit. @shaheedmalik What scenario(s) were you thinking that having the 1s and 10s places' separate would solve? I'm curious if you have a particular need here besides ease of use/nice to have.

@mdtauk the proposal in general was agreed upon as an improvement, but we didn't have the dev resources to act on it, so it was moved "out" and has not been revisited since.
Also, I agree the chevrons are a bit unnecessary, but it's nothing a refreshed template couldn't fix/hide.

Additionally, depending on how the "new" timepicker inherited from ComboBox, it would be a property on the "new" control. Which in this proposal is just a re-skin of ComboBox. If that's what you meant?

**Comment by kmgallahan** [comment]
I like the current design with loop-around scrolling and auto-selection based on the currently highlighted item. It is good for touch and mouse wheels with infinite scroll (where the wheel rotates freely without clicking). I understand though that if your mouse wheel requires a click for each digit change then it could be annoying.

Split digits for minutes/seconds adds extra input steps and clutters the UI. Combo boxes require both scrolling and selection, plus the chevrons are not attractive.

I agree with @mdtauk that accepting keyboard input would be the most helpful addition, and this would not take a step backwards from the current design - depending on what input method is being used.

The main design consideration for keyboard input would be how to indicate which part of the TimerPicker has focus, and how to change focus.

**Comment by mdtauk** [comment]
@kikisaints  With it all going Open Source - that makes the proposal more likely to happen.  WinUI 3.0's initial release is the priority for now, but don't abandon the idea.

![image](https://user-images.githubusercontent.com/7389110/59948329-9c84c500-9467-11e9-8a16-9dd4889442b5.png)

**Comment by kikisaints** [comment]
@mdtauk :smile: Absolutely, just hadn't thought about it until this issue was posted. Although it's outside of the scope of what shaheedmalik is asking for, I did create a separate issue here: https://github.com/microsoft/microsoft-ui-xaml/issues/918

**Comment by jcbeppler** [comment]
This already exist in the UWP:
![image](https://user-images.githubusercontent.com/59994318/100361956-16c15d80-2ffb-11eb-8c94-ab6a52d6ab02.png)

**Comment by shaheedmalik** [comment]
> This already exist in the UWP:
> ![image](https://user-images.githubusercontent.com/59994318/100361956-16c15d80-2ffb-11eb-8c94-ab6a52d6ab02.png)

What are you responding to? That wasn't the purpose of the proposal.
